packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/mod.rs

```diff
@@ -32,8 +33,9 @@ impl<C: AsClient> EntityStore for PostgresStore<C> {
                 .change_context(InsertionError)?,
         );
 
-        let entity_id = EntityId::new(Uuid::new_v4());
+        let entity_id = entity_id.unwrap_or_else(|| EntityId::new(Uuid::new_v4()));
 
+        // TODO: return a better error if the ID already exists
```

Comment:
@TimDiekmann thoughts on how I can do this?

Reply:
```suggestion
        // TODO: match on and return the relevant error
        //   https://app.asana.com/0/1200211978612931/1202574350052904/f
```

Reply:
This should panic when inserting a duplicated key:
```rust
if let Err(report) = transaction.insert_entity_id(entity_id).await {
    if let Some(error) = report.downcast_ref::<tokio_postgres::Error>() {
        if let Some(db_error) = error.as_db_error() {
            if *db_error.code() == SqlState::UNIQUE_VIOLATION {
                panic!("ENTITY ID EXISTS");
            }
        }
    }
}
```

Reply:
I'll add that to the asana task. I think it's better we do this in one concerted effort so that we can actually figure out how to test these paths are working and such :)